adambrg/src/main/java/UserDao.java

```diff
@@ -0,0 +1,142 @@
+
+import org.mindrot.jbcrypt.BCrypt;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class UserDao {
+    private static final String CREATE_USER =
+            "INSERT INTO users(username, email, password) VALUES (?, ?, ? )";
+
+
+    public User create(User user) { //metoda pro vytvareni uzivatele
+        try (Connection conn = DbUtil.getConnection()) {
+            final PreparedStatement statement = conn.prepareStatement(CREATE_USER, PreparedStatement.RETURN_GENERATED_KEYS);
+
+            statement.setString(1, user.getUserName());
+            statement.setString(2, user.getEmail());
+            statement.setString(3, hashPassword(user.getPassword()));
+            statement.executeUpdate();
+
+            final ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                user.setId(resultSet.getInt(1));
+                System.out.println("User ID " + user.getId());
+
+            }
+            return user;
+
+        } catch (SQLException e) {
+            System.err.println("ERROR while creating a user");
+            e.printStackTrace();
+            return null;
+
+        }
+    }
+
+    private static final String UPDATE_USER =
+            "UPDATE users SET username = ?, email = ?, password = ? WHERE id = ?";
+
+    public void update(User user) { //metoda pro aktualizovani inforamci uzivatele
+        try (Connection conn = DbUtil.getConnection()){
+            final PreparedStatement statement = conn.prepareStatement(UPDATE_USER);
+
+            statement.setString(1, user.getUserName());
+            statement.setString(2, user.getEmail());
+            statement.setString(3, hashPassword(user.getPassword()));
```

Comment:
BUG: zaheshovat jen nova hesla, pri aktualizaci stareho nechceme hashovat znova zaheshovane heslo.

```suggestion
statement.setString(3, user.getPassword());
```

Reply:
opraveno